lib/unleash/constraint.rb

```diff
@@ -45,6 +49,10 @@ def matches_context?(context)
     rescue KeyError
       Unleash.logger.warn "Attemped to resolve a context key during constraint resolution: #{self.context_name} but it wasn't \
       found on the context"
+
+      # when the operator is NOT_IN and there is no data, return true. In all other cases the operator doesn't match.
+      return true if operator == :NOT_IN
```

Comment:
Think about it again, this seems to belong in matches_constraint? Method and not in the matches_context?


So for this operator the matches_constraint? Should not throw an exception. (But still should for all other operators).

Also note that this doesn't seem to respect the inverted method parameter. Noting it up the stack should solve that. But maybe add a trivial assert for NOT using the inverted parameter (in addition to the current one using)?

Reply:
The latest commit should solve all of this. The reversing happens in matches_context and in matches_constraint we are now handling the nil values.